staging/src/k8s.io/code-generator/pkg/conversiongen/utils.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conversiongen
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"text/template"
+
+	"k8s.io/gengo/generator"
+	"k8s.io/gengo/namer"
+	"k8s.io/gengo/types"
+	"k8s.io/klog"
+)
+
+// ConversionPair represents a conversion pair from inType to outType
+type ConversionPair struct {
+	InType  *types.Type
+	OutType *types.Type
+}
+
+// A NamedVariable represents a named variable to be rendered in snippets.
```

Comment:
Hm, gengo probably needs to get this concept at some point (not asking for a change).